components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ThrottlingApiServiceImpl.java

```diff
@@ -979,6 +987,48 @@ public Response throttlingBlacklistConditionIdDelete(String conditionId, String
         return null;
     }
 
+    /**
+     * Updates an existing condition status of a blocking condition
+     *
+     * @param conditionId       Id of the block condition
+     * @param body              content to update
+     * @param contentType       Content-Type header
+     * @param ifMatch           If-Match header value
+     * @param ifUnmodifiedSince If-Unmodified-Since header value
+     * @return 200 response if successful
+     */
+    @Override
+    public Response throttlingBlacklistConditionIdPatch(String conditionId, BlockingConditionStatusDTO body,
+                                                        String contentType, String ifMatch, String ifUnmodifiedSince,
+                                                        MessageContext messageContext) {
+        try {
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            String username = RestApiUtil.getLoggedInUsername();
+
+            //This will give BlockConditionNotFoundException if there's no block condition exists with UUID
+            BlockConditionsDTO existingCondition = apiProvider.getBlockConditionByUUID(conditionId);
+            if (!RestApiAdminUtils.isBlockConditionAccessibleToUser(username, existingCondition)) {
+                RestApiUtil.handleAuthorizationFailure(RestApiConstants.RESOURCE_BLOCK_CONDITION, conditionId, log);
```

Comment:
Don't we need to throw `404 - handleResourceNotFoundError` within else part to handle the case if the condition does not exist

Reply:
404 - handleResourceNotFoundError is handled in the catch block

Reply:
If it is a non-existing block condition, does the method throw an APIM exception?